tools/distpartitioning/parmetis_preprocess.py

```diff
@@ -0,0 +1,326 @@
+    return edge_files
+
+def read_node_features(schema_map, tgt_ntype_name, feat_names):
+    """
+    Helper function to read the node features, if present.
+    Only node features which are requested are read from the input dataset. 
+
+    Parameters:
+    -----------
+    schema_map : json dictionary
+        dictionary created by reading the metadata.json file for the input dataset
+	tgt_ntype_name : string
+		node type name, for which node features will be read from the input dataset
+    feat_names : set
+	a set of strings, feature names, which will be read for a given node type	
+
+    Returns:
+    --------
+    dictionary : 
+	a dictionary where key is the feature-name and value is the numpy array
+    """	
+    rank = get_proc_info()
+    node_features = {}
+    if constants.STR_NODE_DATA in schema_map:
+        dataset_features = schema_map[constants.STR_NODE_DATA]
+        if((dataset_features is not None) and (len(dataset_features) > 0)):
+            for ntype_name, ntype_feature_data in dataset_features.items():
+                if ntype_name != tgt_ntype_name : 
+                    continue
+                # ntype_feature_data is a dictionary
+                # where key: feature_name, value: dictionary in which keys are "format", "data".
+                for feat_name, feat_data in ntype_feature_data.items():
+                    if feat_name in feat_names:
+                        my_feat_data_fname = feat_data[constants.STR_DATA][rank]
+                        logging.info(f'Reading: {my_feat_data_fname}')
+                        if (os.path.isabs(my_feat_data_fname)):
+                            node_features[feat_name] = np.load(my_feat_data_fname)
+                        else:
+                            node_features[feat_name] = np.load(os.path.join(input_dir, my_feat_data_fname))
+    return node_features
+
+
+def gen_node_weights_files(schema_map, output):
+    """
+    Function to create node weight files for ParMETIS along with the edge files.
+
+    This function generates node-data files, which will be read by the ParMETIS
+    executable for partitioning purposes. Each line in these files will be of the 
+    following format: 
+    <node_type_id> <weight_list> <type_node_id>
+    where 
+    node_type_id -  is id assigned to the node-type to which a given particular
+    node belongs to
+    weight_list - this is a one-hot vector in which the number in the location of
+    the current nodes' node-type will be set to `1` and other will be `0`
+    type_node_id - this is the id assigned to the node (in the context of the current
+    nodes` node-type). Meaning this id is unique across all the nodes which belong to 
+    the current nodes` node-type.
+
+    Parameters:
+    -----------
+    schema_map : json dictionary
+        dictionary created by reading the metadata.json file for the input dataset
+    output : string
+        location of storing the node-weights and edge files for ParMETIS
+
+    Returns:
```

Comment:
There are 2 returns mentioned here, but only one parameter was returned in L227.